sdk/analytics/src/main/java/com/klaviyo/analytics/state/UserState.kt

```diff
@@ -0,0 +1,146 @@
+package com.klaviyo.analytics.state
+
+import com.klaviyo.analytics.model.ImmutableProfile
+import com.klaviyo.analytics.model.Keyword
+import com.klaviyo.analytics.model.PROFILE_ATTRIBUTES
+import com.klaviyo.analytics.model.Profile
+import com.klaviyo.analytics.model.ProfileKey
+import com.klaviyo.analytics.model.ProfileKey.ANONYMOUS_ID
+import com.klaviyo.analytics.model.ProfileKey.EMAIL
+import com.klaviyo.analytics.model.ProfileKey.EXTERNAL_ID
+import com.klaviyo.analytics.model.ProfileKey.PHONE_NUMBER
+import com.klaviyo.analytics.model.ProfileKey.PUSH_STATE
+import com.klaviyo.analytics.model.ProfileKey.PUSH_TOKEN
+import com.klaviyo.analytics.networking.requests.PushTokenApiRequest
+import com.klaviyo.core.Registry
+import java.util.UUID
+
+/**
+ * Stores information on the currently active user
+ */
+internal class UserState : State {
+
+    private val _externalId = PersistentObservableString(EXTERNAL_ID, ::broadcastChange)
+    override var externalId by _externalId
+
+    private val _email = PersistentObservableString(EMAIL, ::broadcastChange)
+    override var email by _email
+
+    private val _phoneNumber = PersistentObservableString(PHONE_NUMBER, ::broadcastChange)
+    override var phoneNumber by _phoneNumber
+
+    private val _anonymousId = PersistentObservableString(ANONYMOUS_ID, ::broadcastChange) {
+        UUID.randomUUID().toString()
```

Comment:
This is similar to the pattern we used for auto-generating the anonymous ID previously. I'm considering perhaps moving that behavior to the new side effects class, but opted for the lesser change for now.

Reply:
Oh, to elaborate, this method is called as a fallback by `PersistentObservableString` when a value is not currently stored for this key in the persistent store. Thats what we did before too, but all the logic lived inside `UserInfo` as setter/getter functions of the `anonymousId` property.